test/v1/services/CreateAmendForeignServiceSpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import shared.controllers.EndpointLogContext
+import shared.models.domain.{Nino, TaxYear}
+import shared.models.errors._
+import shared.models.outcomes.ResponseWrapper
+import shared.services.ServiceSpec
+import v1.connectors.MockCreateAmendForeignConnector
+import v1.models.request.createAmend.{CreateAmendForeignRequest, CreateAmendForeignRequestBody, ForeignEarnings, UnremittableForeignIncomeItem}
+
+import scala.concurrent.Future
+
+class CreateAmendForeignServiceSpec extends ServiceSpec {
+
+  private val nino    = "AA112233A"
+  private val taxYear = "2019-20"
+
+  private val foreignEarningsModel = ForeignEarnings(
+    customerReference = Some("ref"),
+    earningsNotTaxableUK = 111.11
+  )
+
+  private val unremittableForeignIncomeModel = List(
+    UnremittableForeignIncomeItem(
+      countryCode = "DEU",
+      amountInForeignCurrency = 222.22,
+      amountTaxPaid = Some(333.33)
+    ),
+    UnremittableForeignIncomeItem(
+      countryCode = "FRA",
+      amountInForeignCurrency = 444.44,
+      amountTaxPaid = Some(555.55)
+    )
+  )
+
+  private val amendForeignRequestBody = CreateAmendForeignRequestBody(
+    foreignEarnings = Some(foreignEarningsModel),
+    unremittableForeignIncome = Some(unremittableForeignIncomeModel)
+  )
+
+  val amendForeignRequest: CreateAmendForeignRequest =
+    CreateAmendForeignRequest(nino = Nino(nino), taxYear = TaxYear.fromMtd(taxYear), body = amendForeignRequestBody)
+
+  trait Test extends MockCreateAmendForeignConnector {
+    implicit val logContext: EndpointLogContext = EndpointLogContext("c", "ep")
+
+    val service: CreateAmendForeignService = new CreateAmendForeignService(
+      connector = mockAmendForeignConnector
+    )
+
+  }
```

Comment:
Don't think this Test trait is necessary at all. The CreaetAmendForeignServiceSpec could extend the MockCreateAmendForeignConnector and then the service just be created as a private val. The logContext can be removed entirely